Add spring-pulsar example (#98)
streamnative/examples: spring-pulsar/README.md

# Spring Pulsar Integration Example
This is an example application demonstrating how to use [Spring for Apache Pulsar](https://github.com/spring-projects-experimental/spring-pulsar).

## Prerequisites

> * Java 17 to run 

Reviewer: **[TinyNit]** Adding bullet points may clearly delineate the items

* Maven to compile

> * [StreamNative Cloud](https://streamnative.io) instance or your own Pulsar cluster

Author: It will work on all versions of SN cloud :)

If you are using StreamNative Cloud you need to configure [your pulsar service url and authentication](https://docs.streamnative.io/cloud/stable/connect/overview) in [src/main/resources/application.yaml](src/main/resources/application.yml) as below:

> ```

Reviewer: Add "```yaml" as the source may give nice formatting.

spring:
  pulsar:
    client:
      service-url: pulsar+ssl://free.o-j8r1u.snio.cloud:6651
      auth-plugin-class-name: org.apache.pulsar.client.impl.auth.oauth2.AuthenticationOAuth2
      authentication:
        private-key: file:///Users/user/Downloads/o-j8r1u-free.json
        audience: urn:sn:pulsar:o-j8r1u:free
        issuer-url: https://auth.streamnative.cloud/
```

## Building the example

> You can build the example application by running:

Reviewer 2: ```suggestion
You can build the example application with the following command:
```

```
mvn clean install
```

## Running the example

> You can run the example application by running:

Reviewer 2: ```suggestion
You can run the example application with the following command:
```
